### PR TITLE
o/devicestate: create pending users after receiving a serial

### PR DESCRIPTION
**Pull request by pedronis**
Currently, if you attempt to create a user using a serial-bound assertion before the serial is available, the request will simply fail. This commit makes it so that attempts to create a system-user with a serial-bound assertion fail, but a flag is set within the device manager's state. Once the serial assertion is acquired, any serial-bound users will be created by the device manager's Ensure() function.

This is a reprise of #12852 applying my own feedback there

### CODE REVIEW

File: overlord/devicestate/devicemgr.go [hunk 1]
```diff
@@ -1066,6 +1066,58 @@ func (m *DeviceManager) ensureAutoImportAssertions() error {
...
+
+	var seeded bool
+	if err := m.state.Get("seeded", &seeded); err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+	if !seeded {
+		return nil
+	}
+
+	var waitingOnSerial bool
```
Comment (Meulengracht) [inline overlord/devicestate/devicemgr.go:1085]: Nitpick: wouldn't it make more sense to check this before checking "seeded"? that way we could probably avoid the first call altogether in most cases.

**Review by Meulengracht — APPROVED** [review]
Looks good, thank you for handling this PR, just a small comment

File: overlord/devicestate/devicemgr.go [hunk 2]
```diff
@@ -1066,6 +1066,58 @@ func (m *DeviceManager) ensureAutoImportAssertions() error {
 	return nil
 }
 
+func (m *DeviceManager) ensureSerialBoundSystemUserAssertionsProcessed() error {
```
Comment (alfonsosanchezbeato) [inline overlord/devicestate/devicemgr.go:1069]: I think it would be good for this to have a comment describing the situation when this is useful.

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, a minor comment

File: overlord/devicestate/devicemgr.go (hunk 1 above)
Comment (pedronis) [reply overlord/devicestate/devicemgr.go:1085]: yes, thanks

File: overlord/devicestate/devicemgr.go [hunk 3]
```diff
@@ -1066,6 +1066,58 @@ func (m *DeviceManager) ensureAutoImportAssertions() error {
...
+		return err
+	}
+	if !seeded {
+		return nil
+	}
+
+	var waitingOnSerial bool
+	err := m.state.Get("system-user-waiting-on-serial", &waitingOnSerial)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		logger.Noticef("failed to get serial-bound assert state")
```
[the remark this answers is not in the document]
Comment (pedronis) [reply overlord/devicestate/devicemgr.go:1088]: yes, this was like this in the original branch, but we usually error in this situations

File: overlord/devicestate/devicemgr.go (hunk 2 above)
Comment (pedronis) [reply overlord/devicestate/devicemgr.go:1069]: added comment

**Comment by pedronis** [comment]
I applied feedback and rebased this

**Pull request merged**